text/0000-deprecation.md

```diff
@@ -0,0 +1,111 @@
+- Feature Name: Public Stability
+- Start Date: 2015-09-03
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+
+This RFC proposes to allow library authors to use a `#[deprecated]` attribute,
+with optional `since = "`*version*`"`, `reason = "`*free text*`"` and 
+`use = "`*substitute declaration*`"` fields. The compiler can then
+warn on deprecated items, while `rustdoc` can document their deprecation
+accordingly. 
+
+# Motivation
+
+Library authors want a way to evolve their APIs; which also involves 
+deprecating items. To do this cleanly, they need to document their intentions 
+and give their users enough time to react.
+
+Currently there is no support from the language for this oft-wanted feature
+(despite a similar feature existing for the sole purpose of evolving the Rust
+standard library). This RFC aims to rectify that, while giving a pleasant
+interface to use while maximizing usefulness of the metadata introduced.
+
+# Detailed design
+
+Public API items (both plain `fn`s, methods, trait- and inherent 
+`impl`ementations as well as `const` definitions, type definitions, struct
+fields and enum variants) can be given a `#[deprecated]` attribute. All
+possible fields are optional:
+
+* `since` is defined to contain the exact version of the crate that
+deprecated the item, as defined by Cargo.toml (thus following the semver
+scheme). It makes no sense to put a version number higher than the current
+newest version here, and this is not checked (but could be by external
+lints, e.g. [rust-clippy](https://github.com/Manishearth/rust-clippy).
+To maximize usefulness, the version should be fully specified (e.g. no
+wildcards or ranges).
```

Comment:
'should'? This sentence doesn't add anything if it's not enforced. Is it required or not that this is a full version?